src/azure-cli/azure/cli/command_modules/acs/decorator.py

```diff
@@ -0,0 +1,570 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from knack.log import get_logger
+from typing import Any
+
+from azure.cli.core import AzCommandsLoader
+from azure.cli.core.azclierror import (
+    CLIInternalError,
+    ResourceNotFoundError,
+    ClientRequestError,
+    ArgumentUsageError,
+    InvalidArgumentValueError,
+    MutuallyExclusiveArgumentError,
+    ValidationError,
+    UnauthorizedError,
+)
+from azure.cli.core.commands import AzCliCommand
+from azure.cli.core.profiles import ResourceType
+
+from .custom import (
+    _get_rg_location,
+    _validate_ssh_key,
+    _get_default_dns_prefix,
+    _set_vm_set_type,
+    set_load_balancer_sku,
+    get_subscription_id,
+)
+
+logger = get_logger(__name__)
+
+
+def safe_list_get(li: list, idx: int, default: Any = None):
+    if isinstance(li, list):
+        try:
+            return li[idx]
+        except IndexError:
+            return default
+    return None
+
+
+# pylint: disable=too-many-instance-attributes,too-few-public-methods
+class AKSCreateModels:
+    # Used to store models (i.e. the corresponding class of a certain api version specified by `resource_type`)
+    # which would be used during the creation process.
+    def __init__(
+        self,
+        cmd: AzCommandsLoader,
+        resource_type: ResourceType = ResourceType.MGMT_CONTAINERSERVICE,
+    ):
+        self.cmd = cmd
```

Comment:
self.__cmd

Reply:
Sure, done in commit [7f75d9b](https://github.com/Azure/azure-cli/pull/19259/commits/7f75d9be8f02597676dae06627b1c90cbe801329).